File: python3/350.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1Map, nums2Map = defaultdict(int), defaultdict(int)
        res = []
        for num in nums1:
            nums1Map[num] += 1
        for num in nums2:
            nums2Map[num] += 1
        for k, v in nums1Map.items():
            for i in range(min(v, nums2Map[k])):
                res.append(k)
        return res

File: golang/350.go
func intersect(nums1 []int, nums2 []int) []int {
    // count occurences of each number in nums1
    map1 := make(map[int]int)
    for _, num := range nums1 {
        map1[num] += 1
    }

    // compare values in nums2 to map1
    res := []int{}
    for _, num := range nums2 {
        if map1[num] > 0 {
            res = append(res, num)
            map1[num]--
        }
    }
    return res
}
